File: 1502-construct-k-palindrome-strings/construct-k-palindrome-strings.java
class Solution {
    public boolean canConstruct(String s, int k) {
        int n = s.length();
        if(n<k)return false;
        if(n==k) return true;

        int[] freq = new int[26];
        int oddCount =0;
        for(char ch : s.toCharArray()){
            freq[ch-'a']++ ;
        }

        for(int i=0; i<26; i++){
            if(freq[i]%2==1)oddCount++ ; 
        }

        if(oddCount>k)return false;
        return true;
    }
}

File: 1502-construct-k-palindrome-strings/construct-k-palindrome-strings.py
class Solution:
    def canConstruct(self, s: str, k: int) -> bool:

        if len(s)<k:
            return False
        
        if len(s)==k:
            return True

        freq = [0]*26
        for i in s : 
            freq[ord(i) - ord('a')] += 1
        
        oddCount = sum(1 for i in freq if i%2==1)

        return oddCount <= k 

        